doc: Add VMware Tools Mention & Highlight Known Issues (#850)
harvester/docs: docs/advanced/addons/vmimport.md

The `storageClass` field specifies the [StorageClass](../storageclass.md) to be used for images and provisioning persistent volumes during the import process. If not specified, the default StorageClass will be used.

> By default, the vm-import-controller attempts to graceful shut down the guest operating system of the source virtual machine before starting the import process. If the virtual machine is not gracefully shut down within a specific period, a hard power off is forced. You can adjust this time period for the graceful shutdown by changing the value of the `gracefulShutdownTimeoutSeconds` field, which is set to `60` seconds by default. A hard power off without attempting a graceful shutdown can be forced by setting the `forcePowerOff` field to `true`. To note, the graceful shutdown via VMware based VMs can only be obtained by having "VMware Tools" installed on the virtual machine that you are attempting to import.  If "VMware Tools" is not installed on the source VM that you are importing and graceful shutdown is preferred, the vm-import-controller will likely have logs that resemble something like: `handler virtualmachine-import-job-change: failed to shutdown the guest OS of the source VM: ServerFaultCode: Cannot complete operation because VMware Tools is not running in this virtual machine., requeuing` , this is an expected behavior.

Author: @jillian-maroket thanks for the review  :smile: :+1: - I'll sign off and commit the suggestion

> #### Known issues
> * **Source virtual machine name is not RFC1123 compliant**: When creating a virtual machine object, the vm-import-controller add-on uses the name of the source virtual machine, which may not meet the Kubernetes object [naming criteria](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names). You may need to rename the source virtual machine to allow successful completion of the import.

Reviewer: ```suggestion
#### Known Issues

##### Source Virtual Machine Name Is Not RFC1123-Compliant

When creating a virtual machine object, the vm-import-controller add-on uses the name of the source virtual machine, which may not meet the Kubernetes object [naming criteria](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names). You may need to rename the source virtual machine to allow successful completion of the import.
```

Reviewer: I need to change the structure to improve readability.

> * **v1.6.0-rc3 vm-import-controller gracefulShutdownTimeoutSeconds, once it is exceeded a forceful shutdown is not performed and VM never migrates from VMware**: When creating a VirtualMachineImport and using a VMware source, if VMware Tools are not present on the VM a graceful shutdown can not be completed, the VM will not migrate. You will be required to one of the following as a work-around:

Reviewer 2: ```suggestion
* **v1.6.0 vm-import-controller gracefulShutdownTimeoutSeconds, once it is exceeded a forceful shutdown is not performed and VM never migrates from VMware**: When creating a VirtualMachineImport and using a VMware source, if VMware Tools are not present on the VM a graceful shutdown can not be completed, the VM will not migrate. You will be required to one of the following as a work-around:
```

Author: great catch :smile: - I've updated and forced pushed up the changes :+1:

> * **v1.6.0 vm-import-controller gracefulShutdownTimeoutSeconds, once it is exceeded a forceful shutdown is not performed and VM never migrates from VMware**: When creating a VirtualMachineImport and using a VMware source, if VMware Tools are not present on the VM a graceful shutdown can not be completed, the VM will not migrate. You will be required to one of the following as a work-around:
>   * shutdown the VM yourself prior to migration to Harvester 
>   * provide the `forcePowerOff: true` in your VirtualMachineImport spec 
>   * install VMware Tools, or [open-vm-tools, as highlighted in this Broadcom knowledge base doc](https://knowledge.broadcom.com/external/article?legacyId=2073803)

Reviewer: ```suggestion
##### VMware-Based Virtual Machine Without VMware Tools Is Not Migrated

When you attempt to import a VMware-based virtual machine in Harvester v1.6.0, the following occur if [VMware Tools](https://knowledge.broadcom.com/external/article/315382/overview-of-vmware-tools.html) is not installed on the virtual machine:

- The vm-import-controller does not gracefully shut down the guest operating system.
- When the graceful shutdown period (`gracefulShutdownTimeoutSeconds`) lapses, the vm-import-controller does not force a hard poweroff.
- The virtual machine is not migrated from VMware.

To address the issue, perform one of the following workarounds:

- Shut down the virtual machine before migrating it to Harvester 
- In the `VirtualMachineImport` CRD spec, set the `forcePowerOff` field to `true`.  
- Install VMware Tools or [open-vm-tools](https://knowledge.broadcom.com/external/article?legacyId=2073803).
```

Author: @jillian-maroket thanks for the review  :smile: :+1: - I'll sign off and commit the suggestion
